src/Symfony/Component/Yaml/Inline.php

```diff
@@ -627,7 +627,7 @@ public static function evaluateBinaryScalar($scalar)
 
     private static function isBinaryString($value)
     {
-        return preg_match('/[^\x09-\x0d\x20-\xff]/', $value);
+        return !preg_match('//u', $value);
```

Comment:
maybe we should include also strings that contains control chars (with a few excluded line CR, LF, TAB)?